tests/dialects/test_teradata.py

```diff
@@ -210,3 +210,73 @@ def test_cast(self):
                 "teradata": "TRYCAST('-2.5' AS DECIMAL(5, 2))",
             },
         )
+
+    def test_time(self):
+        self.validate_identity("SELECT CURRENT_TIMESTAMP")
+
+        self.validate_all(
+            "CURRENT_TIMESTAMP",
+            read={
+                "snowflake": "CURRENT_TIMESTAMP()",
+            },
+            write={
+                "teradata": "CURRENT_TIMESTAMP",
+                "snowflake": "CURRENT_TIMESTAMP()",
+            },
+        )
+
+        self.validate_all(
+            "SELECT DATE_ADD('2023-01-01', 5, YEAR)",
```

Comment:
This isn't valid Teradata syntax, right? Let's use valid syntax for the first argument, because tests like these can be confusing. We can use `read` to show how `DATE_ADD` is transpiled _to_ Teradata.

Reply:
Ditto about all other test cases that do this.

Reply:
Makes sense, I'll update these